File: core-java/src/main/java/chessgame/domain/model/Bishop.java
package chessgame.domain.model;

import java.util.Set;

/**
 * Bishop of Chess game.
 *
 * @author Prestes, E. M.
 * @see chessgame.domain.Piece
 * @since September 2016
 */
public class Bishop extends AbstractPiece {

    /**
     * Bishop's constructor.
     *
     * @param board Board informed.
     * @param color Color informed.
     * @see BoardPosition
     * @see PieceColor
     */
    public Bishop(Board board, PieceColor color) {
        super(board, color);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public Set<BoardPosition> getAvailablePositions() {
        final BoardPosition myPosition = getPosition();
        return collectDiagonals(myPosition);
    }
}

File: core-kotlin/src/main/kotlin/chessgame/domain/model/Bishop.kt
package chessgame.domain.model

class Bishop(
        override val board: Board,
        override val color: PieceColor) : AbstractPiece(board, color) {

    override fun getAvailablePositions(): Set<BoardPosition> {
        val availablePositions = mutableSetOf<BoardPosition>()

        collectDiagonals(position, availablePositions)

        return availablePositions
    }
}